### PR TITLE
Add StreamListener usage sample

### LINKED ISSUE #8: Add sample for StreamListener support
A demo application that showcases how to do message mapping using @StreamListener annotation.

### PR DESCRIPTION
**Pull request by ilayaperumalg**
This resolves #8

### CODE REVIEW

File: stream-listener/src/main/java/demo/Converters.java [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+		}
+	}
+
+	public static class Bar {
+
+		public Bar(String value) {
+			this.value = value;
+		}
+
+		private String value = "init";
```
Comment (garyrussell) [inline stream-listener/src/main/java/demo/Converters.java]: Convention is to declare fields first.

File: stream-listener/src/main/java/demo/Converters.java [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+			try {
+				if (message.getPayload() instanceof Foo) {
+					Foo fooPayload = (Foo) message.getPayload();
+					result = new Bar(fooPayload.getValue());
+				}
+			}
+			catch (Exception e) {
+				logger.error(e.getMessage(), e);
+				return null;
+			}
```
Comment (garyrussell) [inline stream-listener/src/main/java/demo/Converters.java]: Is this the suggested convention? (eat the exception and return null).

Shouldn't we throw an `o.s.messaging...MessageConversionException` ?

Comment (ilayaperumalg) [inline stream-listener/src/main/java/demo/Converters.java]: I agree. Created https://github.com/spring-cloud/spring-cloud-stream/issues/456 to fix this in SCSt.

**Comment by ilayaperumalg** [comment]
Addressed the comments, rebased and updated the PR.

**Comment by garyrussell** [comment]
Squashed, merged as 2414a2f1ef1035adaa230f1a7a5632f2882417ff

**Pull request closed without merging**